log/loki.go

```diff
@@ -128,15 +163,47 @@ func LokiFromConfigLine(ctx context.Context, fallbackLogger logrus.FieldLogger,
 				continue
 			}
 
-			return nil, fmt.Errorf("unknown loki config key %s", key)
+			return fmt.Errorf("unknown loki config key %s", key)
 		}
 	}
 
-	h.client = &http.Client{Timeout: h.pushPeriod}
+	return nil
+}
 
-	go h.loop()
+func parseArray(key, value string, i int, args []string) ([]string, int, error) {
+	if value == "" {
+		return []string{}, i, nil
+	}
+	var ended bool
+	var result []string
+	if value[0] != '[' {
+		return nil, 0, fmt.Errorf("%s needs to be in the form [value1,value2,...,valueN]", key)
+	}
+	if value[len(value)-1] == ']' {
+		value = value[:len(value)-1]
+		ended = true
+	}
+	value = value[1:]
+
+	result = append(result, value)
+	for i++; i < len(args) && !ended; i++ {
+		value = args[i]
+		if value == "" {
+			return nil, 0, fmt.Errorf("array item in %s need to not be empty", key)
+		} else if value[len(value)-1] == ']' {
+			value = value[:len(value)-1]
+			ended = true
+		}
+		if value == "" {
+			return nil, 0, fmt.Errorf("array item in %s need to not be empty", key)
+		}
+		result = append(result, value)
+	}
+	if !ended {
+		return nil, 0, fmt.Errorf("%s needs to be in the form [value1,value2,...,valueN], but it didn't end", key)
```

Comment:
Couldn't you also check if `value[len(value)-1] != ']'` in the initial check on line 179 and return the same message from there?

Reply:
I would still need this check though as I do need to go through the rest of the args (see https://github.com/loadimpact/k6/pull/1639/files#r495747384) to see when the array ends.